bot/cogs/moderation.py

```diff
@@ -193,6 +199,104 @@ async def clear(self, ctx: Context, amount: int) -> None:
         await asyncio.sleep(2.5)
         await message.delete()
 
+    @command()
+    @has_permissions(manage_messages=True)
+    async def shift(self, ctx, count: int, target: TextChannel, copy: bool = False) -> None:
+        """Copy or Move specified messages amount to specified channel"""
+        messages = []
+        async for message in ctx.channel.history(limit=count):
+            embed = Embed(description=message.content, color=Color.green())
+            embed.set_author(name=message.author.name, icon_url=message.author.avatar_url)
+            embed.timestamp = message.created_at
+            messages.append(embed)
+
+            if not copy:
+                await message.delete()
+
+        await target.send(f'Message source : {ctx.channel.mention}.')
+
+        for embed in reversed(messages):
+            await target.send(embed=embed)
+
+    @command()
+    @has_permissions(administrator=True)
+    async def dm(self, ctx: Context, members: Greedy[t.Union[Member, Role]], *, message: str) -> None:
+        """Dm a List of Specified User from Your Guild."""
+        embed = Embed(
+            title="Notice!",
+            description=message,
+            color=Color.dark_magenta()
+        )
+        embed.set_footer(text=f"From {ctx.guild.name}", icon_url=ctx.guild.icon_url)
+
+        for member in members:
+            if isinstance(member, Role):
+                print("works")
+                for mem in member.members:
+                    print("for")
+                    await mem.send(embed=embed)
+                print("above return")
+                return
+            else:
+                await member.send(embed=embed)
+
+    @command()
+    @has_permissions(administrator=True)
+    async def dmall(self, ctx: Context, *, message: str) -> None:
+        """Dm all Users from Your Guild."""
+        embed = Embed(
+            title="Notice!",
```

Comment:
Don't hardcore titles

Reply:
we can't use dynamic titles

Reply:
if we do this : `async def dmall(self, ctx: Context, *, title: str, *, message: str) -> None:`
which is gonna be title, and which one is message?

Reply:
Something like this:
```
>>dmall title="Important mail" description="Desripto"

Reply:
uh, using kwargs in bot commands?
not possible